### PR TITLE
Math: Updates to Point2D and Point3D

### PR DESCRIPTION
**Pull request by marcusnaslund**
Part 1 of #491

### CODE REVIEW

File: source/math/FloatPoint3D.ooc [hunk 1]
```diff
@@ -35,12 +35,17 @@ FloatPoint3D: cover {
 	init: func@ (=x, =y, =z)
 	init: func@ ~default { this init(0.0f, 0.0f, 0.0f) }
 	init: func@ ~fromPoint2D (point: FloatPoint2D, z := 0.0f) { this init(point x, point y, z) }
+	pNorm: func (p: Float) -> Float {
+		p == 1 ?
+		this x abs() + this y abs() + this z abs() :
+		(this x abs() pow(p) + this y abs() pow(p) + this z abs() pow(p)) pow(1.0f / p)
```
Comment (davidhesselbom) [inline source/math/FloatPoint3D.ooc]: These two lines would look better if you indent them one step.

File: source/math/FloatPoint3D.ooc [hunk 2]
```diff
@@ -35,12 +35,17 @@ FloatPoint3D: cover {
...
+	}
 	scalarProduct: func (other: This) -> Float { this x * other x + this y * other y + this z * other z }
 	vectorProduct: func (other: This) -> This { This new(this y * other z - other y * this z, -(this x * other z - other x * this z), this x * other y - other x * this y) }
-	spherical: func (radius, azimuth, elevation: Float) -> This {
+	spherical: static func (radius, azimuth, elevation: Float) -> This {
 		This new(radius * (azimuth cos()) * (elevation sin()), radius * (azimuth sin()) * (elevation sin()), radius * (elevation cos()))
 	}
-	angles: func (rx, ry, n: Float) -> This {
+	angles: static func (rx, ry, n: Float) -> This {
 		z := n*n sqrt() / (1 + ry tan() squared() + rx tan() squared())
```
Comment (davidhesselbom) [inline source/math/FloatPoint3D.ooc]: Wierd, I thought magic would catch this...

File: source/math/FloatPoint3D.ooc (hunk 1 above)
Comment (marcusnaslund) [inline source/math/FloatPoint3D.ooc]: Putting everything on the same line is shorter than line 42, but maybe it isn't clear enough.

Comment (davidhesselbom) [inline source/math/FloatPoint3D.ooc]: Yeah, no, that line would be illegible. Just do this
```ooc
p == 1 ?
	this x abs() + this y abs() + this z abs() :
	(this x abs() pow(p) + this y abs() pow(p) + this z abs() pow(p)) pow(1.0f / p)
```

Comment (marcusnaslund) [inline source/math/FloatPoint3D.ooc]: Fixed here and in the 2D version.

**Pull request merged**